### PR TITLE
feat(types): support define custom payload

### PR DESCRIPTION
**Pull request by uinz**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: jwt.test-d.ts [hunk 1]
```diff
@@ -55,22 +55,31 @@ expectAssignable<Function>(app.jwt.verify)
...
-    let data = await app.jwt.verify(token);
-    const user = req.user;
-    reply.send({ token });
+  const token = app.jwt.sign({ user: "userName" });
+  let data = await app.jwt.verify(token);
+  const user = req.user;
+  reply.send({ token });
 });
+
+// define custom payload
```
Comment (Ethan-Arrowood) [inline jwt.test-d.ts:78]: could you create a new testing file that has some declaration merging and tests for it?

**Review by Ethan-Arrowood — CHANGES_REQUESTED** [review]

File: jwt.test-d.ts (hunk 1 above)
Comment (uinz) [reply jwt.test-d.ts:78]: Declaring the module will pollute the global type environment, I have not found a good way to use dts to test

Comment (fox1t) [reply jwt.test-d.ts:78]: Do we really care about adding types to the global type env? I think we should test this properly.

**Comment by fox1t** [comment]
Well done. I like this PR!

File: jwt.test-d.ts (hunk 1 above)
Comment (Ethan-Arrowood) [reply jwt.test-d.ts:78]: Good point, this is good to go then

**Review by Ethan-Arrowood — APPROVED** [review]

**Review by fox1t — APPROVED** [review]

**Pull request merged**